### PR TITLE
Read more non-standard starttime fields

### PR DESCRIPTION
**Pull request by hofaflo**

### CODE REVIEW

File: edfio/_header_field.py [hunk 1]
```diff
@@ -150,7 +150,11 @@ def __init__(self, length: int, *, is_settable: bool = False) -> None:
         super().__init__(length, is_settable=is_settable)
 
     def decode(self, field: bytes) -> datetime.time:
-        hours, minutes, seconds = (int(n) for n in decode_str(field).split("."))
+        time = decode_str(field)
+        match = DATE_OR_TIME_PATTERN.fullmatch(time.replace(" ", ""))
+        if match is None:
+            raise ValueError(f"Invalid time for format hh.mm.ss: {time!r}")
+        hours, minutes, seconds = (int(g) for g in match.groups())
```
Comment (marcoross) [inline edfio/_header_field.py:157]: Nice solution using capture groups.

File: tests/test_faq.py [hunk 2]
```diff
@@ -36,6 +40,20 @@ def test_q2_date_decode_different_formats(field: bytes):
...
 
 
+@pytest.mark.parametrize(
+    "field",
+    [
+        b"02.08.51",
+        b"2.8.51  ",
+        b"2. 8.51 ",
+        b"02:08-51",
+        b"02/08'51",
```
Comment (marcoross) [inline tests/test_faq.py:50]: Add b"3 5 98" to test spaces as separators?

**Review by marcoross — COMMENTED** [review]
Since we are trying to allow all kinds of crazy separators in the date and time fields. However, we could test what happens if the numbers are outside a reasonable range. For example, if the minutes/seconds are not between 0 and 59 or the hours not between 0 and 23. What if the day exceeds the number of days in the selected month or the month is outside the interval 1..12?
This questions might be more relevant when we try to fix broken EDF headers.

File: edfio/_header_field.py [hunk 3]
```diff
@@ -116,20 +129,7 @@ def __init__(self, length: int, *, is_settable: bool = False) -> None:
...
-            {separator}
-            {one_or_two_digits}  # month
-            {separator}
-            {one_or_two_digits}  # year
-            """,
-            re.VERBOSE,
-        )
-        date = date.replace(" ", "")
-        match = re.fullmatch(pattern, date)
+        match = DATE_OR_TIME_PATTERN.fullmatch(date.replace(" ", ""))
```
[the remark this answers is not in the document]
Comment (hofaflo) [reply edfio/_header_field.py]: good idea, done!

**Comment by hofaflo** [comment]
> we could test what happens if the numbers are outside a reasonable range. For example, if the minutes/seconds are not between 0 and 59 or the hours not between 0 and 23. What if the day exceeds the number of days in the selected month or the month is outside the interval 1..12?

There are already tests for invalid month or day in [`tests/test_header_field.py`](https://github.com/the-siesta-group/edfio/blob/1461a9fbe23624d95f027b8642d2e0dcc8e8602d/tests/test_header_field.py#L58-L65), but we could add some for seconds/minutes/hours? 

> This questions might be more relevant when we try to fix broken EDF headers.

Do you mean we could try to reorder day/month/year or hour/minute/second to see if one (and only one) combination is valid?

**Comment by marcoross** [comment]
> There are already tests for invalid month or day in [`tests/test_header_field.py`](https://github.com/the-siesta-group/edfio/blob/1461a9fbe23624d95f027b8642d2e0dcc8e8602d/tests/test_header_field.py#L58-L65), but we could add some for seconds/minutes/hours?

That's probably not necessary.

> 
> > This questions might be more relevant when we try to fix broken EDF headers.
> 
> Do you mean we could try to reorder day/month/year or hour/minute/second to see if one (and only one) combination is valid?

No, that would go too far. I was just thinking that when/if we implement an auto-fix option, we should re-consider these tolerances when reading. Because what this effectively does is implement an "auto-fix on access" for date fields without actually fixing the EDF file.

**Review by marcoross — APPROVED** [review]

**Pull request merged**